drivers/can/can_stm32h7_fdcan.c

```diff
@@ -124,6 +127,22 @@ static int can_stm32h7_clock_enable(const struct device *dev)
 		}
 	}
 
+	/* Check if clock has correct range according to chosen regulator voltage
+	 * scaling (Table 62 of RM0399 Rev 4):
+	 */
+	ret = clock_control_get_rate(clk,
+		(clock_control_subsys_t)&stm32h7_cfg->pclken[1], &fdcan_clock);
```

Comment:
stm32h7_cfg->pclken[1] is not necessarily defined. You should check first with `if (IS_ENABLED(STM32H7_FDCAN_DOMAIN_CLOCK_SUPPORT) && (stm32h7_cfg->pclk_len > 1))` like above if you're going to use it. But then you still have to treat the case where the default clock is used (HSE in this case).

Reply:
Thanks. It is valid. I will put it under the hood of the above if statement. 
HSE will be newer bigger than 125mhz, thus is always valid for vso0

Reply:
You can add a comment about default HSE clock, indicating that there is nothing to do. It will prevent questions later when someone will read this code again ;)

Reply:
Upated as requested.